pandas/tests/indexes/test_base.py

```diff
@@ -1228,13 +1251,21 @@ def test_equals_op_mismatched_multiindex_raises(self, index):
         with pytest.raises(ValueError, match="Lengths must match"):
             df.index == index
 
-    def test_equals_op_index_vs_mi_same_length(self):
+    def test_equals_op_index_vs_mi_same_length(self, using_infer_string):
         mi = MultiIndex.from_tuples([(1, 2), (4, 5), (8, 9)])
         index = Index(["foo", "bar", "baz"])
-
-        result = mi == index
-        expected = np.array([False, False, False])
-        tm.assert_numpy_array_equal(result, expected)
+        if using_infer_string:
+            import pyarrow as pa
+
+            with pytest.raises(pa.lib.ArrowNotImplementedError, match="has no kernel"):
+                with tm.assert_produces_warning(
+                    DeprecationWarning, check_stacklevel=False
```

Comment:
Is this our DeprecationWarning?

Reply:
let me remove this, I have a pr aligning the differences in behaviour here